core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/NoOpMetricCollector.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.metrics;
+
+import software.amazon.awssdk.annotations.SdkPreviewApi;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * A metric collector that doesn't do anything.
+ *
+ * <b>NOTE:</b> This is a Preview API and is subject to change so it should not be used in production.
+ */
+@SdkPreviewApi
+@SdkPublicApi
+public final class NoOpMetricCollector implements MetricCollector {
+    private static final NoOpMetricCollector INSTANCE = new NoOpMetricCollector();
+
+    private NoOpMetricCollector() {
+    }
+
+    @Override
+    public String name() {
+        return "NoOp";
+    }
+
+    @Override
+    public <T> void reportMetric(SdkMetric<T> metric, T data) {
+    }
+
+    @Override
+    public MetricCollector createChild(String name) {
+        throw new UnsupportedOperationException("No op collector does not support createChild");
+    }
+
+    @Override
+    public MetricCollection collect() {
+        throw new UnsupportedOperationException("No op collector does not support collect");
+    }
```

Comment:
We might want to consider supporting these, so that enabling the "NoOpMetricCollector" won't cause all of the SDK calls to fail with UnsupportedOperationException.